src/textual/widgets/_markdown.py

```diff
@@ -0,0 +1,773 @@
+        color: $text-muted;
+        margin: 1 0;
+    }
+
+    """
+
+
+class Paragraph(Block):
+    """A paragraph Markdown block."""
+
+    DEFAULT_CSS = """
+    Markdown > Paragraph {
+         margin: 0 0 1 0;
+    }
+    """
+
+
+class BlockQuote(Block):
+    """A block quote Markdown block."""
+
+    DEFAULT_CSS = """
+    BlockQuote {
+        background: $boost;
+        border-left: outer $success;
+        margin: 1 0;
+        padding: 0 1;
+    }
+    BlockQuote > BlockQuote {
+        margin-left: 2;
+        margin-top: 1;
+    }
+
+    """
+
+
+class BulletList(Block):
+    """A Bullet list Markdown block."""
+
+    DEFAULT_CSS = """
+
+    BulletList {
+        margin: 0;
+        padding: 0 0;
+    }
+
+    BulletList BulletList {
+        margin: 0;
+        padding-top: 0;
+    }
+
+    """
+
+
+class OrderedList(Block):
+    """An ordered list Markdown block."""
+
+    DEFAULT_CSS = """
+
+    OrderedList {
+        margin: 0;
+        padding: 0 0;
+    }
+
+    OrderedList OrderedList {
+        margin: 0;
+        padding-top: 0;
+    }
+
+    """
+
+
+class Table(Block):
+    """A Table markdown Block."""
+
+    DEFAULT_CSS = """
+    Table {
+        margin: 1 0;
+    }
+    Table > DataTable {
+        width: 100%;
+        height: auto;
+    }
+    """
+
+    def compose(self) -> ComposeResult:
+        def flatten(block: Block) -> Iterable[Block]:
+            for block in block.blocks:
+                if block.blocks:
+                    yield from flatten(block)
+                yield block
+
+        headers: list[Text] = []
+        rows: list[list[Text]] = []
+        for block in flatten(self):
+            if isinstance(block, TH):
+                headers.append(block.text)
+            elif isinstance(block, TR):
+                rows.append([])
+            elif isinstance(block, TD):
+                rows[-1].append(block.text)
+
+        table: DataTable = DataTable(zebra_stripes=True, show_cursor=False)
+        table.can_focus = False
+        table.add_columns(*headers)
+        table.add_rows([row for row in rows if row])
+        yield table
+        self.blocks.clear()
+
+
+class TBody(Block):
+    """A table body Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class THead(Block):
+    """A table head Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class TR(Block):
+    """A table row Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class TH(Block):
+    """A table header Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class TD(Block):
+    """A table data Markdown block."""
+
+    DEFAULT_CSS = """
+
+    """
+
+
+class Bullet(Widget):
+    """A bullet widget."""
+
+    DEFAULT_CSS = """
+    Bullet {
+        width: auto;
+    }
+    """
+
+    symbol = reactive("●​ ")
+
+    def render(self) -> Text:
+        return Text(self.symbol)
+
+
+class ListItem(Block):
+    """A list item Markdown block."""
+
+    DEFAULT_CSS = """
+
+    ListItem {
```

Comment:
Will this clash with the `ListItem` widget used in `ListView` in any way?

Reply:
Good point, it might. Should probablt be something like `Markdown > ListItem`

Reply:
I think we should rename `ListItem` here too (`MarkdownListItem`?), since it's public and could catch users out. If someone targets `ListItem` in their CSS, we don't want there to be ambiguity.